### ISSUE TITLE
Crash when recording certain windows

### ISSUE THREAD

**Issue by CarwynNelson** [popey/vokoscreenng-snap#1]
## Description

It looks like the screen recorder crashes when recording certain windows. At the moment I have only tried this on two applications:
* `Ubuntu Software Center` - Crashes vokoscreen-ng
* `Tillix Terminal`. - Records just fine

I am using the stable channel (3.0.1 2020-02-06).

## Steps to reproduce
1. Open the vokoscreenNG snap
2. Open the Ubuntu Software Center
3. In vokoscreen select "window" and press "start"
4. Select the ubuntu software center window that you just opened
5. Record some actions and press "stop" in vokoscreen
6. vokoscreen will become unresponsive and you will soon be prompted to "Force Quit"

## System Specifications

OS: Ubuntu 19.10 x86_64
Kernel: 5.3.0-42-generic
DE: Gnome 3.34.3
Display System: X11

## Debug Output

<details>
<summary>
I launched the vokoscreen-ng snap via the terminal and captured 
the following debug output:
</summary>

```
Qt: Session management error: None of the authentication protocols specified are supported
propsReply "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.1125\" (uid=1000 pid=15752 comm=\"/snap/vokoscreen-ng/33/vokoscreenNG \" label=\"snap.vokoscreen-ng.vokoscreen-ng (enforce)\") interface=\"org.freedesktop.DBus.Properties\" member=\"GetAll\" error name=\"(unset)\" requested_reply=\"0\" destination=\"org.freedesktop.NetworkManager\" (uid=0 pid=1186 comm=\"/usr/sbin/NetworkManager --no-daemon \" label=\"unconfined\")"
nmReply "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.1125\" (uid=1000 pid=15752 comm=\"/snap/vokoscreen-ng/33/vokoscreenNG \" label=\"snap.vokoscreen-ng.vokoscreen-ng (enforce)\") interface=\"org.freedesktop.NetworkManager\" member=\"GetDevices\" error name=\"(unset)\" requested_reply=\"0\" destination=\"org.freedesktop.NetworkManager\" (uid=0 pid=1186 comm=\"/usr/sbin/NetworkManager --no-daemon \" label=\"unconfined\")"
"Object path cannot be empty"
[vokoscreenNG] Version: 3.0.1
[vokoscreenNG] Locale: en_GB
[vokoscreenNG] Log from: 2020-03-28 17:01:57
[vokoscreenNG] Country: UnitedKingdom
[vokoscreenNG] Qt: 5.9.5
[vokoscreenNG] GStreamer 1.14.5
[vokoscreenNG] Operating system: Ubuntu Core 18
[vokoscreenNG] CPU Architecture: x86_64
[vokoscreenNG] vokoscreenNG running as: xcb client
[vokoscreenNG] vokoscreenNG running on: x11
[vokoscreenNG] Desktop: ubuntu:GNOME
[vokoscreenNG] Icon-Theme: Adwaita
[vokoscreenNG] Styles: (Windows, Fusion)
[vokoscreenNG] Qt-PluginsPath:      /usr/lib/x86_64-linux-gnu/qt5/plugins
[vokoscreenNG] Qt-TranslationsPath: /usr/share/qt5/translations
[vokoscreenNG] Qt-LibraryPath:      /usr/lib/x86_64-linux-gnu
[vokoscreenNG] Settings: /home/carwyn/snap/vokoscreen-ng/33/.config/vokoscreenNG/vokoscreenNG.ini
[vokoscreenNG] Log: /home/carwyn/snap/vokoscreen-ng/33/.config/vokoscreenNG/log/2020_03_28_17_01_54.log
[vokoscreenNG] CompositingManager running: true

[vokoscreenNG] [PulseAudio] Found: Monitor of Built-in Audio Analogue Stereo Device: alsa_output.pci-0000_00_1f.3.analog-stereo.monitor
[vokoscreenNG] [PulseAudio] Found: Built-in Audio Analogue Stereo Device: alsa_input.pci-0000_00_1f.3.analog-stereo
[vokoscreenNG] [PulseAudio] Found: Monitor of LE-Carwyn's Bose Nose Device: bluez_sink.4C_87_5D_CA_72_B7.a2dp_sink.monitor

[vokoscreenNG] Symbols: + available, - not available
[vokoscreenNG] + matroskamux
[vokoscreenNG] + webmmux
[vokoscreenNG] + avimux
[vokoscreenNG] + mp4mux
[vokoscreenNG] + qtmux
[vokoscreenNG] + x264enc
[vokoscreenNG] - openh264enc
[vokoscreenNG] - vaapih264enc
[vokoscreenNG] - vaapimpeg2enc
[vokoscreenNG] + vp8enc
[vokoscreenNG] + vorbisenc
[vokoscreenNG] + flacenc
[vokoscreenNG] + opusenc
[vokoscreenNG] + lamemp3enc

[vokoscreenNG] + ximagesrc
[vokoscreenNG] + pulsesrc
[vokoscreenNG] + queue
[vokoscreenNG] + capsfilter
[vokoscreenNG] + videoconvert
[vokoscreenNG] + videorate
[vokoscreenNG] + audioconvert
[vokoscreenNG] + audiorate
[vokoscreenNG] + filesink
[vokoscreenNG] + videoscale
[vokoscreenNG] + h264parse
[vokoscreenNG] + adder

[vokoscreenNG] Name from screen:  eDP-1-1
[vokoscreenNG] Screen available desktop width : 1920
[vokoscreenNG] Screen available desktop height: 1080
[vokoscreenNG] DevicePixelRatio: 1  (Normal displays is 1, Retina display is 2)
[vokoscreenNG] Vertical refresh rate of the screen in Hz: 240
[vokoscreenNG] Screen orientation Qt::ScreenOrientation(LandscapeOrientation)
[vokoscreenNG] Color depth of the screen:  24
[vokoscreenNG] Model from screen:  
[vokoscreenNG] Manufactur from screen:  
[vokoscreenNG] SerialNumber from screen:  
[vokoscreenNG] ItemText in Combobox: eDP-1-1 :  1920 x 1080
[vokoscreenNG] ItemData in Combobox: x=0 y=0 with=1920 height=1080


[vokoscreenNG] ExtensionsPath: /home/carwyn/snap/vokoscreen-ng/33/vokoscreen_extensions/
[vokoscreenNG] No library for extension found
 
[vokoscreenNG] Desktop session is a X11 session
radioButtonScreencastFullscreen=false
radioButtonScreencastWindow=true
radioButtonScreencastArea=false
checkBoxCameraOnOff=false
checkBoxCameraMirrorVertical=false
checkBoxCameraMirrorHorizontal=false
checkBoxCameraInvert=false
checkBoxCameraGray=false
checkBoxCameraMono=false
checkBoxCameraWindowFrame=false
checkBoxShowInSystray=true
checkBoxShowInSystrayAlternative=true
checkBoxMinimizedWhenRecordingStarts=false
checkBoxStartsMinimized=false
checkBoxResetAtNextStart=false
checkBoxLookForUpdates=true
checkBoxMouseCursorOnOff=false
checkboxAudioDevice-0=false
checkboxAudioDevice-1=false
checkboxAudioDevice-2=false
checkBoxScale=false
checkBoxMagnifier=false
comboBoxCamera=
comboBoxFormat=mkv
comboBoxVideoCodec=x264
comboBoxx264Preset=superfast
comboBoxAudioCodec=vorbis
comboBoxOnlineHelp=Automatically
comboBoxAreaSize=320 x 200 CGA 16 : 10
comboBoxScale=320 x 200 CGA 16 : 10
comboBoxScreencastScreen=eDP-1-1 :  1920 x 1080
sliderVolume=70
sliderVideo=0
sliderCameraWindowSize=2
sliderHour=0
sliderMinute=0
sliderStopRecordingAfterHouers=0
sliderStopRecordingAfterMinutes=0
sliderStopRecordingAfterSeconds=15
sliderLimitOfFreeDiskSpace=250
sliderShowInSystrayAlternative=48
sliderSecondWaitBeforeRecording=1
sliderX264=17
sliderOpenh264=23
sliderFrames=25
sliderScreencastCountDown=0
sliderMagnification=2
lineEditVideoPath=/home/carwyn/Videos
[vokoscreenNG] Start record with: ximagesrc display-name=:1 use-damage=false show-pointer=true xid=79691786 ! capsfilter caps=video/x-raw,framerate=25/1 ! videoconvert ! videorate ! x264enc qp-min=17 qp-max=17 speed-preset=superfast threads=12 ! video/x-h264, profile=baseline ! matroskamux name=mux writing-app=vokoscreenNG_3.0.1 ! filesink location=/home/carwyn/Videos/vokoscreenNG-2020-03-28_17-02-21.mkv
 
gst-launch-1.0 -e \
    ximagesrc display-name=:1 use-damage=false show-pointer=true xid=79691786 \
        ! capsfilter caps=video/x-raw,framerate=25/1 \
        ! videoconvert \
        ! videorate \
        ! x264enc qp-min=17 qp-max=17 speed-preset=superfast threads=12 \
        ! video/x-h264, profile=baseline \
        ! matroskamux name=mux writing-app=vokoscreenNG_3.0.1 \
        ! filesink location=/home/carwyn/Videos/vokoscreenNG-2020-03-28_17-02-21.mkv

[1]    15752 killed     vokoscreen-ng
```

</details>

**Comment by CarwynNelson** [comment]
tldr; it looks like this has been fixed in a recent commit. The bug happens when compiling 3.0.1 outside of snap as well.

I decided to do a bit of additional testing to see if I could replicate the bug in other environments, or narrow down the issue.

I installed the snap on a quickemu vm of ubuntu-mate 19.10 with gnome-software installed via apt and could replicate the crash.
I then decided to compile version `3.0.1` from source inside the vm and found that the crash does not occur.
During both of these tests I can still record full screen and area video just fine.

When compiling the latest commit from "master" (called 3.0 in upstream) I don't seem to get this bug, so maybe this has been fixed in a recent commit?

I do get another weird bug with the video where I get a black screen where I can only see a cursor with trails, but since I get this with the snap and with master I imagine this is a separate bug when recording gnome-software as a window.
